### ISSUE TITLE
FN for S3257: Primitive types should be omitted from initialized or defaulted declarations

### ISSUE THREAD

**Issue by jccollet** [SonarSource/SonarTS#173]
In the following code the issue is raised on 's2' but not 's':
```
function GenerateString() : string {
  return "Hello World"
}

let s : string = GenerateString();
let s2 : string = "Hello World";

```

**Comment by stas-vilchik** [comment]
This rule is a tslint one ([no-inferrable-types](https://palantir.github.io/tslint/rules/no-inferrable-types/)). To fix the false negative, we need to rewrite the rule implementation. Because we didn't plan to do it for 1.0, let's postpone to 1.1.

**Comment by vilchik-elena** [comment]
I don't agree that it's a FN. The point of the rule is to avoid putting types where it's obvious. From `let s : string = GenerateString();` it's not obvious that `s` has type `string` (to know that, you should look deeper in the code)

**Comment by varora1406** [comment]
@vilchik-elena  I understand your point.
My requirement of opting typescript is that static code analysis should take care that types are not getting changed. If we can achieve it optionally, that's also good. Maybe many people would need it.

This FN talks about same I guess so.
This Sonar rule is not checking if my type is changed.
But TSLint rule checks that, although it requires above rule of Sonar to get disabled then.

Are we bringing something in SonarRules close to what you see as my requirement?
